src/rules/jsxBanPropsRule.ts

```diff
@@ -45,37 +51,31 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new JsxBanPropsWalker(sourceFile, this.getOptions()));
-    }
-}
-
-class JsxBanPropsWalker extends Lint.RuleWalker {
-    private isInJsxAttribute = false;
-    private bannedProps: Map<string, string>;
-
-    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
-        super(sourceFile, options);
-
-        const propsToBan = options.ruleArguments;
-        this.bannedProps = propsToBan
-            ? new Map<string, string>(propsToBan.map((prop: string[]): [string, string] =>
+        const bannedProps = this.ruleArguments
+            ? new Map<string, string>(this.ruleArguments.map((prop: string[]): [string, string] =>
                 [prop[0], prop.length > 1 ? prop[1] : ""]))
             : new Map<string, string>();
+        return this.applyWithFunction(sourceFile, walk, { bannedProps });
     }
+}
 
-    protected visitJsxAttribute(node: ts.JsxAttribute) {
-        this.isInJsxAttribute = true;
-        super.visitJsxAttribute(node);
-        this.isInJsxAttribute = false;
-    }
+function walk(ctx: Lint.WalkContext<IRuleOptions>): void {
+    let inJsxAttribute = false;
 
-    protected visitIdentifier(node: ts.Identifier) {
-        if (this.isInJsxAttribute) {
-            const propName = (node as ts.Identifier).text;
-            if (this.bannedProps.has(propName)) {
-                const propBanExplanation = this.bannedProps.get(propName);
-                this.addFailureAtNode(node, Rule.FAILURE_STRING_FACTORY(propName, propBanExplanation));
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        if (isJsxAttribute(node)) {
+            inJsxAttribute = true;
+            ts.forEachChild(node, cb);
+            inJsxAttribute = false;
```

Comment:
would be better to use a different callback to recurse here instead of explicitly keeping track of this state flag